app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterSourceSync.kt

```diff
@@ -128,14 +128,20 @@ fun syncChaptersWithSource(
             var now = Date().time
 
             for (i in toAdd.indices.reversed()) {
-                val c = toAdd[i]
-                c.date_fetch = now++
-                // Try to mark already read chapters as read when the source deletes them
-                if (c.isRecognizedNumber && c.chapter_number in deletedReadChapterNumbers) {
-                    c.read = true
-                }
-                if (c.isRecognizedNumber && c.chapter_number in deletedChapterNumbers) {
-                    readded.add(c)
+                val chapter = toAdd[i]
+                chapter.date_fetch = now++
+                if (chapter.isRecognizedNumber && chapter.chapter_number in deletedChapterNumbers) {
+                    // Try to mark already read chapters as read when the source deletes them
+                    if (chapter.chapter_number in deletedReadChapterNumbers) {
+                        chapter.read = true
+                    }
+                    // Try to to use the fetch date it originally had to not pollute 'Updates' tab
+                    toDelete.filter { it.chapter_number == chapter.chapter_number }
+                        .minByOrNull { it.date_fetch }!!.let {
```

Comment:
you're already using let, no need for the !! that'll just lead to crashes
```suggestion
                        .minByOrNull { it.date_fetch }.let {
```

Reply:
I took this feature from main tachiyomi so I didn't change much of it but yeah you are totally right, I'll replace it with a `?`